src/map.rs

```diff
@@ -876,6 +879,39 @@ impl<K: Ord + Default, V: Default, const N: usize> SgMap<K, V, N> {
     pub fn len(&self) -> usize {
         self.bst.len()
     }
+
+    /// Gets the given key's corresponding entry in the map for in-place manipulation.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// use scapegoat::SgMap;
+    ///
+    /// let mut count = SgMap::<&str, usize, 10>::new();
+    ///
+    /// // count the number of occurrences of letters in the vec
+    /// for x in vec!["a", "b", "a", "c", "a", "b"] {
+    ///     *count.entry(x).or_insert(0) += 1;
+    /// }
+    ///
+    /// assert_eq!(count["a"], 3);
+    /// ```
+    pub fn entry(&mut self, key: K) -> Entry<'_, K, V, N> {
+        use crate::tree::Idx;
+        use crate::tree::node::NodeGetHelper;
+
+        let ngh: NodeGetHelper<Idx> = self.bst.priv_get(None, &key);
+        if let Some(node_idx) = ngh.node_idx() {
```

Comment:
Minor nitpick: this `if let Some(_) ... else` could be an exhaustive `match` for `Some(_)` or `None`. B/c `if let` is typically used to ignore the `None` case, so here it adds a little complexity.

The rest of the code looks great! :relaxed: